### ISSUE TITLE
File picker not showing on directory open

### ISSUE THREAD

**Issue by bbrodriges** [helix-editor/helix#8515]
### Summary

File picker behaviour has been broken somewhere between tag `23.05` and commit `1852292451d984de33ea526b3b7dc7f51c6130eb`. On the most recent commit file picker does not show when invoke Helix with directory path as argument.

### Reproduction Steps

1. Build Helix from source at `1852292451d984de33ea526b3b7dc7f51c6130eb`
2. Invoke `hx /somedir/someproject`, no file picker will be shown
3. Invoke `cd /somedir/someproject && hx .`, file picker will be shown

On `23.05` release both invocations would show file picker.

### Helix log

```
2023-10-11T12:43:10.371 helix_vcs [INFO] Error {
    context: "failed to open git repo",
    source: Discover(
        InaccessibleDirectory {
            path: "<path_to_non-git_project>",
        },
    ),
}
2023-10-11T12:43:10.372 helix_vcs [INFO] failed to open diff base for <path_to_non-git_project>
2023-10-11T12:43:10.372 helix_vcs [INFO] Error {
    context: "failed to open git repo",
    source: Discover(
        InaccessibleDirectory {
            path: "<path_to_non-git_project>",
        },
    ),
}
2023-10-11T12:43:10.372 helix_vcs [INFO] failed to obtain current head name for <path_to_non-git_project>
```

### Platform

macOS

### Terminal Emulator

iTerm2 Build 3.4.21

### Helix Version

23.05-473-g18522924

**Comment by the-mikedavis** [comment]
Looks like a regression from https://github.com/helix-editor/helix/pull/8498 (https://github.com/helix-editor/helix/pull/8498#issuecomment-1756997071)

**Comment by woojiq** [comment]
Off topic: @bbrodriges you don't need to format the commit hash with `.
Just paste as is and github will generate a nice formatted hash with the link for you. 
E.g. 5cb76e74f9bd226f1b4757f478bf49873d9ecfe6

**Comment by bjorn-ove** [comment]
It is fixed in #8520
